network/interface.go

```diff
@@ -47,20 +47,19 @@ type Transactions interface {
 	// Reprocess walks the DAG and publishes all transactions matching the contentType via Nats
 	// This is an async process and will not return any feedback
 	Reprocess(contentType string)
+	// WithPersistency returns a SubscriberOption for persistency. It allows the DAG KVStore to be used as persistent store for notifications.
+	// The notifications will then have ACID properties
+	WithPersistency() SubscriberOption
```

Comment:
weird API design, low level function mixed in between high-level (Engine-level) functions.

I'd say make it an uncorrelated `SubscriberOption` like the rest and sort it out in the Go code how to get a handle to the KVStore when processing the options.

Reply:
Not possible since the option MUST pass the same DB as the dag KVStore. This DB is only accessible from within the network. It used to be able to pass it but then other parts of the Node where accessing the network/data KVStore....